### PR TITLE
Updated future warning msg in transform() for Series.groupby

### LINKED ISSUE #55066: BUG: Update FutureWarning message to be more clear
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
df = pd.DataFrame({"ID": [1,1,2,2,3],"B": [1,10,5,5,3]})
df_new = df[df.groupby(["ID"])["B"].transform(max) == df["B"]]
```

### Issue Description

This isn't necessarily a bug, but the other issue categories didn't fit it either.

In #53974, the use of built-in callable functions for `SeriesGroupBy.transform()`was deprecated.  The current `FutureWarning` message that is displayed is:
```
E       FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass 'max' instead.
/pandas/core/apply.py:1828: FutureWarning
```
While the change is reasonable, I think the warning message could be updated to better convey the message. When I first saw the warning message, I didn't realize what was wrong. Obviously using the built-in functions was no longer supported, but the message said `to keep current behavior pass 'max' instead`, however this is exactly what I was doing, but I was passing the callable `max` instead.

After some time of reading the release notes and documentation, I realized the warning message was referring to the string "max".   

### Expected Behavior

My proposal is to just update the `FutureWarning` produced from this to explicitly state a string should be passed. Updating 
```
FutureWarning: .... To keep current behavior pass 'max' instead.
```
to
```
FutureWarning: .... To keep current behavior pass the string 'max' instead.
```

### Installed Versions

2.1.0

### PR DESCRIPTION
**Pull request by rsm-23**
- [ ] closes #55066 
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by RyanBerger98** [comment]
Simple change to make the warning message a lot more clear, thank you!

**Comment by rsm-23** [comment]
@rhshadrach can you please review this?

File: pandas/core/apply.py [hunk 1]
```diff
@@ -1832,12 +1832,12 @@ def warn_alias_replacement(
         full_alias = alias
     else:
         full_alias = f"{type(obj).__name__}.{alias}"
-        alias = f"'{alias}'"
+        alias = f'`"{alias}"`'
```
Comment (mroeschke) [inline pandas/core/apply.py:1835]: Might be good to remove the ``` ` ```s as they might not always render like markdown for example

Comment (rsm-23) [reply pandas/core/apply.py]: Updated @mroeschke

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rsm-23

**Pull request merged**